### ISSUE TITLE
HiOp Examples 

### ISSUE THREAD

**Issue by nychiang** [LLNL/hiop#418]
Since @pelesh  mentioned the redesign of sparse examples, I think it is also a good idea to reorder/rename the hiop examples like this
```
Example -+--- Dense  -+-- EX1
         |            |
         |            +-- Ex2
         |            |
         |            ...
         |           
         +---  MDS   -+-- EX1
         |            |
         |            +-- Ex2
         |            |
         |            ...
         |           
         +--- Sparse -+-- EX1
         |            |
         |            +-- Ex2
         |            |
         |            ...
         |           
         +--- PriDec -+-- EX1          
                      |
                      +-- Ex2
                      |
                      ...
```
This allow us to add a new example to the desired linear system easily and to keep the name sorted.

@cnpetra

**Comment by cnpetra** [comment]
this file organization is neat. @pelesh any comments?

**Comment by pelesh** [comment]
> this file organization is neat. @pelesh any comments?

I agree. This is also a good time to refactor CMake and move test examples definitions from the root `CMakeLists.txt` file into a hierarchical structure.

**Comment by cnpetra** [comment]
yes, it would be a good time indeed. This is refactoring you mentioned recently? I was under the impression that you meant refactoring the example driver code. But CMake refactoring makes more sense...

**Comment by pelesh** [comment]
There are actually two suggestions I made regarding this:
1. Refactor CMake in hierarchical fashion, so when adding a new example you need to edit only CMake file from that group of examples.
2. Improve command line input options so that one can run multiple tests from the same source/binary. For example, in Sparse group, run examples with Cholesky, LU (on both GPU and CPU) and LDL on CPU by running the same binary only different command line inputs. This is I believe a minor modification to examples code.

The latter, however, could trigger some refactoring deeper in the HiOp code. For example, there is no HiOp option to use Cholesky solver. As far as I can see, Cholesky is selected implicitly when GPU is enabled and condensed KKT system is selected.

**Comment by cnpetra** [comment]
closed by PR #420
